src/features/machine/computer.js

```diff
@@ -232,21 +130,11 @@ function buildTrackLoop(state, i, t) {
 
 // ensure vertices do not exceed machine boundary limits, and endpoints as needed
 export const polishVertices = (state, vertices) => {
-  const machine = state.machine
+  vertices = vertices.map(vertex => Victor.fromObject(vertex))
+  const machineClass = state.machine.rectangular ? RectMachine : PolarMachine
 
   if (vertices.length > 0) {
-    if (machine.rectangular) {
-      if (machine.rectOrigin.length === 1) {
-        vertices = addRectEndpoints(machine, vertices)
-      }
-
-      const sizeX = (machine.maxX - machine.minX)/2.0
-      const sizeY = (machine.maxY - machine.minY)/2.0
-      vertices = enforceRectLimits(vertices, sizeX, sizeY)
-    } else {
-      vertices = addPolarEndpoints(machine, vertices)
-      vertices = enforcePolarLimits(vertices, machine.maxRadius)
-    }
+    vertices = new machineClass(vertices, state.machine).polish().vertices
```

Comment:
Is this common in javascript? Just make a temporary class, call the interesting function on it, grab what you want out of it, and throw it in the garbage :). I like it.